src/orion/benchmark/task/profet/profet_task.py

```diff
@@ -0,0 +1,556 @@
+        task_id = task_id or 0
+        super().__init__(
+            task_id=task_id,
+            rng=rng,
+            seed=seed,
+            max_trials=max_trials,
+            fixed_dims=kwargs,
+        )
+        self.benchmark = benchmark
+        if not input_dir:
+            input_dir = Path(os.environ.get("DATA_DIR")) / "profet"
+        else:
+            input_dir = Path(input_dir)
+
+        if checkpoint_dir is None:
+            checkpoint_dir = input_dir / "checkpoints"
+        # TODO: Save the train config in the `configuration` of the task.
+        self.train_config = train_config or MetaModelTrainingConfig()
+        task_hash_params = dict(
+            benchmark=benchmark,
+            task_id=task_id,
+            rng=rng,
+            seed=seed,
+            **asdict(self.train_config),
+        )
+        task_hash = compute_identity(**task_hash_params)
+
+        filename = f"{task_hash}.pkl"
+
+        checkpoint_file = Path(checkpoint_dir) / filename
+        logger.debug(f"Checkpoint file for this task: {checkpoint_file}")
+
+        # The config for the training of the meta-model.
+        # TODO: This should probably be a part of the hash for the "id" of the task!
+        self.train_config = train_config
+
+        if os.path.exists(checkpoint_file):
+            self.net, self.h = load_task_network(checkpoint_file)
+        else:
+            logger.info(
+                f"Checkpoint file {checkpoint_file} doesn't exist: re-training the meta-model."
+            )
+            # TODO: Figure out where/how to set the logger config.
+            logger.info(f"Task hash params: {task_hash_params}")
+            print(f"Task hash params:", json.dumps({k: str(v) for k, v in task_hash_params.items()}, indent="\t"))
+
+            checkpoint_file.parent.mkdir(exist_ok=True, parents=True)
+            # Need to re-train the meta-model and sample this task.
+            self.net, self.h = get_task_network(
+                input_dir, benchmark, rng, task_id, config=train_config
+            )
+            save_task_network(checkpoint_file, benchmark, self.net, self.h)
+
+    @property
+    def hash(self) -> str:
+        # TODO: Return a unique "hash"/id/key for this task
+
+        if is_dataclass(self):
+            return compute_identity(**asdict(self))
+        return compute_identity(
+            benchmark=self.benchmark,
+            task_id=self.task_id,
+            **self.fixed_values,
+            seed=self.seed,
+        )
+
+    @property
+    def space(self) -> Dict[str, str]:
+        space = spaces[self.benchmark].copy()
+        for k in self.fixed_values:
+            if k in space:
+                space.pop(k)
+        return space
+
+    @property
+    def full_space(self) -> Dict[str, str]:
+        return spaces[self.benchmark]
+
+    def __call__(self, x: np.ndarray = None, **kwargs):
+        if x is not None:
+            assert not kwargs, "can't pass x and kwargs at the same time!"
```

Comment:
It may be more convenient to define one variable that is used when both are passed, with a warning stating that both were passed but only x was used for instance.

Reply:
Only using `x` from now on. Just seems simpler imo.

Reply:
Actually need to also have **kwargs because of how the tasks get called by the benchmark.

Reply:
ok!